db/batchlog_manager.cc

```diff
@@ -140,7 +140,13 @@ future<> db::batchlog_manager::stop() {
     }
     _stop = true;
     _timer.cancel();
+    // Synchronize with do_batch_log_replay.
+    // Although _sem might be held only on shard 0
+    // it is safe to wait for it on all shards.
+    return _sem.wait().then([this] {
```

Comment:
Maybe teach `do_batchlog_replay()` run inside the `_gate` instead? Thus the `_gate.close()` will capture those too.

Reply:
solved in a better way in v2 by waiting on _started future returned from the replay loop after requesting abort.